tests/test_schemas.py

```diff
@@ -0,0 +1,172 @@
+            ],
+            "sms:validationRules": ["Rule_1", "Rule_2"],
+        }
+
+        assert expected == actual
+
+    def test_attribute_exists(self, helpers):
+
+        se_obj = helpers.get_schema_explorer("example.model.jsonld")
+
+        # test when attribute is present in data model
+        attribute_present = df_parser.attribute_exists(se_obj, "Patient")
+
+        # test when attribute is not present in data model
+        attribute_absent = df_parser.attribute_exists(se_obj, "RandomAttribute")
+
+        assert attribute_present == True
+        assert attribute_absent == False
+
+    def test_check_schema_definition(self, helpers):
+
+        data_model_csv_path = helpers.get_data_path("example.model.csv")
+
+        example_model_df = pd.read_csv(data_model_csv_path)
+
+        # when all required headers are provided in the CSV data model
+        actual_df = df_parser.check_schema_definition(example_model_df)
+
+        assert actual_df is None
+
+        # when either "Requires" or "Requires Component" is present
+        # in column headers, raise ValueError
+        if "DependsOn Component" in example_model_df.columns:
+            del example_model_df["DependsOn Component"]
+
+            example_model_df["Requires Component"] = ""
+
+            with pytest.raises(ValueError):
+                assert df_parser.check_schema_definition(example_model_df)
+
+    def test_create_nx_schema_objects(self, helpers):
+
+        se_obj = helpers.get_schema_explorer("example.model.jsonld")
+
+        # path to extended CSV data model which has one additional attribute
+        # namely, "Assay"
+        extended_csv_model_path = helpers.get_data_path("example_extension.model.csv")
+
+        extended_model_df = pd.read_csv(extended_csv_model_path)
+
+        extended_csv_model_se = df_parser.create_nx_schema_objects(
+            extended_model_df, se_obj
+        )
+
+        # check if the "Assay" attribute has been added to the new SchemaExplorer 
+        # object with attributes from the extended schema
+        result = df_parser.attribute_exists(extended_csv_model_se, "Assay")
+
+        assert result is True
+
+    def test_get_base_schema_path(self):
+
+        base_schema_path = "/path/to/base_schema.jsonld"
+
+        # path to base schema is returned when base_schema is passed
+        result_path = df_parser._get_base_schema_path(base_schema=base_schema_path)
+
+        assert result_path == "/path/to/base_schema.jsonld"
+
+        # path to default BioThings data model is returned when no
+        # base schema path is passed explicitly
+        biothings_path = df_parser._get_base_schema_path()
+
+        assert os.path.basename(biothings_path) == "biothings.model.jsonld"
+
+    def test_convert_csv_to_data_model(self, helpers):
+
+        csv_path = helpers.get_data_path("example.model.jsonld")
+
+        extended_csv_model_path = helpers.get_data_path("example_extension.model.csv")
```

Comment:
I see that you do this twice in this particular test suite. In that case, I recommend that you abstract the Pandas code I suggested you write in an earlier comment into a fixture at the top of this file.

Reply:
Then, you would store the file into a temporary CSV file (look into `tempfile` and the `with` statement) before reloading using `_convert_csv_to_data_model()`.

Reply:
Yup, that makes sense. I've created a module level fixture called `extended_schema_path()` which appends the new `Assay` attribute (for testing) to the example schema.